improvements to sms test view

snopoke opened the pull request:
## Product Description
This view is 'hidden' and requires accessing the URL directly. It is also not documented anywhere public.

The changes in this PR:

* allow specifying phone number via form instead of the URL
* allow simulating which gateway the message came from
* support feature flags like 'inbound leniency'

With these changes it is possible to test more complex SMS workflows without needing a real gateway:
1. Create a gateway for the domain using the 'Test' gateway type
    - This prevents messages actually going out
    - It also means that any feature flags enabled for the domain will be activated during message processing since the gateway is owned by the domain. This is not true when using a global gateway.
2. (optional) Share the gateway with other domains
3. Either
    - Set the new gateway to the default for the domain(s)
    - OR create a new `SQLMobileBackendMapping` via the Django admin to selectively use the backend
4. Trigger an SMS workflow and use the 'Test SMS' view to simulate incoming messages

## Safety Assurance

### Safety story
Changes to a hidden view.

### Automated test coverage
None

### QA Plan
None

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

snopoke committed:
split method to allow calling with raw args

snopoke committed:
update message_test view to make it more useful

* allow specifying phone number via form
* allow specifying backend (and use backend when simulating message)
* support feature flags like 'inbound leniency'

snopoke committed:
set backend message ID

snopoke committed:
better error message

orangejenny commented:
> This view is 'hidden' and requires accessing the URL directly.

Could it go in the Messaging menu but behind an `is_superuser` check?

snopoke committed:
add 'claim this channel' feature

snopoke committed:
add to messaging nav for superusers and support

snopoke commented:
@orangejenny I've added a commit to put this in the nav for superusers and if the 'support' flag is enabled.

I've also added a commit that allows the user the attempt a 'claim' of the SMS channel when the 'multiple contacts, one phone number' feature is enabled.

orangejenny commented on corehq/apps/sms/forms.py:
```diff
@@ -1327,6 +1327,10 @@ class SentTestSmsForm(Form):
         label=gettext_lazy("Gateway"),
         required=False
     )
+    claim_channel = BooleanField(
```
cool
